src/app.js

```diff
@@ -74,3 +72,86 @@ function allocateColor (region) {
     }
 }
 
+
+/////////////////////// OPDRACHT 2/////////////////////////////////////////////
+
+// sla de referentie naar het formulier op en plaats er een submit-event listener op
+const searchForm = document.getElementById('search-form');
+//koppel eventListener eraan. Als submit wordt getriggerd dan wordt de functie searchCountry uitgvoerd
+searchForm.addEventListener('submit', searchCountry );
+
+const printCountry = document.getElementById('search-result');
+const errorMessageBox = document.getElementById('error-message');
+
+// maak functie search country. Wanneer event object wordt aangemaakt wordt parameter meegegeven
+function searchCountry(e){
+   //voorkom standaard actie, dus dat de pagina ververst
+    e.preventDefault();
+    //koppel searchValue aan html element om de waarde op te halen. IK SNAP DEZE STAP NIET!
+    const searchValue = document.getElementById("search-value");
+   //ingevoerde waarde wordt in de fetchdata methode toegevoegd
+    fetchData(searchValue.value);
+    //legen van het invoerveld
+    searchValue.value = '';
+}
+
+
+//hierboven heette het searchValue.value en heet vanaf nu country
+async function fetchData (country){
+    //Hiermee weet ik dat na elke zoekopdracht de printing leeg wordt gemaakt
+    printCountry.innerHTML = ``;
+    errorMessageBox.innerHTML = ``;
+
+    try{
+        //backticks for string interpolation
+        const result = await axios.get(`https://restcountries.com/v2/name/${country}`)
+        const chosenCountry = result.data[0];
```

Comment:
Je kunt hier nog spreading toepassen, zodat je niet overal `chosenCountry.` vóór hoeft te zetten. Voorbeeldje:
```suggestion
        const { name, subregion, population, capital, currencies, languages } = result.data[0];
```